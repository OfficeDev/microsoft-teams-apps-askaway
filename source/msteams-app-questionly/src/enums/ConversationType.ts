/**
 * Define value for conversation type in teams app.
 */
export enum ConversationType {
    Channel = 'channel',
}
