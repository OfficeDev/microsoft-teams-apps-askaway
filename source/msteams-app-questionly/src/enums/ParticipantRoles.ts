/**
 * Define value for participant role in a meeting.
 */
export enum ParticipantRoles {
    Presenter = 'Presenter',
    Organizer = 'Organizer',
    Attendee = 'Attendee',
}
