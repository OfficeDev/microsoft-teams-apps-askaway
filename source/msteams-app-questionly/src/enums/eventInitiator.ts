/**
 * Define value for different event initiators.
 */
export enum EventInitiator {
    MainCard = 'MainCard',
    RestApi = 'RestApi',
}
