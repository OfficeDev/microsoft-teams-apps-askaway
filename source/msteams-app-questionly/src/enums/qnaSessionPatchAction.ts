/**
 * Patch actions supported for qna session.
 */
export enum QnaSessionPatchAction {
    End = 'end',
}
