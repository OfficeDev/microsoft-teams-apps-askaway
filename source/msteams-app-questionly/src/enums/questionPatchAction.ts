/**
 * Patch actions supported for question.
 */
export enum QuestionPatchAction {
    Upvote = 'upvote',
    Downvote = 'downvote',
    MarkAnswered = 'markAnswered',
}
