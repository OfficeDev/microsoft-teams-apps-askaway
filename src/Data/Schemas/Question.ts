import * as mongoose from 'mongoose';

const QuestionSchema = new mongoose.Schema({
    amaSessionId: {
        type: mongoose.SchemaTypes.ObjectId,
        ref: 'AMASession',
        required: true,
    },
    userId: {
        type: String,
        ref: 'User',
        required: true,
    },
    content: {
        type: String,
        required: true,
        minlength: 1,
        trim: true,
    },
    voters: [{ type: String, ref: 'User' }],
    dateTimeCreated: {
        type: Date,
        default: Date.now,
    },
});

/**
 * Exports the Question schema model for external use.
 */
export const Question = mongoose.model('Question', QuestionSchema);
