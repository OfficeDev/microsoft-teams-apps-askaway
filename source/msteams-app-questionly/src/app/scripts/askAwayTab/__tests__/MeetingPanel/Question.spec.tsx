/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import { Card, Flex, Button, Text, Avatar } from '@fluentui/react-northstar';
import Adapter from 'enzyme-adapter-react-16';
import Question from '../../MeetingPanel/Question';
import { ParticipantRoles } from '../../../../../enums/ParticipantRoles';

configure({ adapter: new Adapter() });

describe('AskAwayTabRemove Component', () => {
    const question = {
        id: '',
        sessionId: '',
        content: '',
        dateTimeCreated: new Date(),
        isAnswered: false,
        author: { id: '', name: '' },
        votesCount: 0,
        voterAadObjectIds: [],
    };
    const isUserLikedQuestion = false;
    const renderHoverElement = null;
    const questionId = '';
    const questionTab = '';
    const userId = '';
    const userRole = ParticipantRoles.Presenter;
    const onClickAction = jest.fn();

    it('should match the snapshot', () => {
        const wrapper = shallow(
            <Question
                question={question}
                isUserLikedQuestion={isUserLikedQuestion}
                renderHoverElement={renderHoverElement}
                questionId={questionId}
                questionTab={questionTab}
                userId={userId}
                userRole={userRole}
                onClickAction={onClickAction}
            />
        );
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(
            <Question
                question={question}
                isUserLikedQuestion={isUserLikedQuestion}
                renderHoverElement={renderHoverElement}
                questionId={questionId}
                questionTab={questionTab}
                userId={userId}
                userRole={userRole}
                onClickAction={onClickAction}
            />
        );

        expect(component.find(Card)).toHaveLength(1);
        expect(component.find(Text)).toHaveLength(3);
        expect(component.find(Flex)).toHaveLength(3);
        expect(component.find(Avatar)).toHaveLength(1);
        expect(component.find(Button)).toHaveLength(1);
    });
});
