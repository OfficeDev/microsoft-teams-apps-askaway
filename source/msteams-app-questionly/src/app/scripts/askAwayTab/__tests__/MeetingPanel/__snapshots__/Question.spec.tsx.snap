// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskAwayTabRemove Component should match the snapshot 1`] = `
<div
  className="card-divider"
  key=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Card
    accessibility={[Function]}
    aria-roledescription="card avatar"
    className="card-layout"
    size="medium"
  >
    <CardHeader
      fitted={true}
    >
      <Flex
        gap="gap.small"
      >
        <Avatar
          getInitials={[Function]}
          name=""
          size="smaller"
        />
        <Flex>
          <Text
            as="span"
            className="author-name"
            content=""
            weight="regular"
          />
          <Flex
            className="like-icon"
            vAlign="center"
          >
            <Button
              accessibility={[Function]}
              as="button"
              className="like-icon-size"
              disabled={true}
              icon={
                <LikeIcon
                  outline={true}
                />
              }
              iconOnly={true}
              onClick={[Function]}
              size="medium"
              text={true}
            />
            <Text
              as="span"
              content={0}
            />
          </Flex>
        </Flex>
      </Flex>
    </CardHeader>
    <CardBody>
      <Text
        as="span"
        className="card-body-question"
        content=""
      />
    </CardBody>
  </Card>
</div>
`;
